src/main/java/de/maxhenkel/audioplayer/FilenameMappings.java

```diff
@@ -0,0 +1,87 @@
+package de.maxhenkel.audioplayer;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.UUID;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.NbtAccounter;
+import net.minecraft.nbt.NbtIo;
+import net.minecraft.server.MinecraftServer;
+
+public class FilenameMappings {
+
+    public static void append(MinecraftServer server, String filename, UUID sound) {
+        URI location = server.getWorldPath(AudioManager.AUDIO_DATA).resolve("filename_mappings.dat").toUri();
+        File mappingsFile = new File(location);
+        String extension = filename.substring(filename.length() - 4);
+
+        try {
+            CompoundTag root = loadOrDefault(mappingsFile);
+            ListTag mappings = getMappingListOrDefault(root);
+
+            CompoundTag entry = new CompoundTag();
+            entry.putString("uuid", sound.toString() + extension);
+            entry.putString("filename", filename);
+
+            mappings.add(entry);
+
+            if (!root.contains("mappings")) {
```

Comment:
Why does the root compound need a nested list? Couldn't the root be the list?

Reply:
This might be a oversight, In the nbt library i normally use for projects you have to use a compound tag as the root so thats why i did it that way, force of habit. but maybe Mojangs inbuilt library allows a list to be the root.